brainframe/api/stubs/capsules.py

```diff
@@ -28,6 +31,63 @@ def get_capsules(self, timeout=DEFAULT_TIMEOUT) -> List[Capsule]:
         capsules, _ = self._get_json(req, timeout)
         return [Capsule.from_dict(d) for d in capsules]
 
+    def load_capsule(self, storage_id: int,
+                     source_path: Optional[Path] = None,
+                     timeout: float = DEFAULT_TIMEOUT) -> Capsule:
+        """Loads and initializes a capsule from capsule data in storage.
+
+        :param storage_id: The ID of the raw capsule data in storage
+        :param source_path: If available, the path to the source code on the
+            development machine can be provided. Doing so allows stack traces
+            to point to the correct source location.
+        :param timeout: The timeout to use for this request
+        :return: The loaded capsule
+        """
+        req = f"/api/plugins"
+        req_object = {
+            "storage_id": storage_id,
+            "dev_options": {
+                "source_path": str(source_path),
+            },
+        }
+        capsule = self._put_json(req, timeout, json.dumps(req_object))
+        return Capsule.from_dict(capsule)
+
+    def upload_and_load_capsule(self, data: Union[bytes, BinaryIO],
+                                source_path: Optional[Path] = None,
+                                timeout: float = DEFAULT_TIMEOUT) -> Capsule:
+        """Uploads capsule data to storage, then loads and initializes a
+        capsule from it. This is a utility method that combines the
+        functionality of `new_storage` and `load_capsule`.
+
+        :param data: The data to store, either as bytes or as a file-like
+        :param source_path: If available, the path to the source code on the
+            development machine can be provided. Doing so allows stack traces
+            to point to the correct source location.
+        :param timeout: The total timeout to use for all requests
+        :return: The loaded capsule
+        """
+        request_start = time()
+        storage_id = self.new_storage(data,
+                                      mime_type="application/octet-stream",
+                                      timeout=timeout)
+        remaining_time = timeout - (time() - request_start)
+
+        return self.load_capsule(storage_id,
+                                 source_path,
+                                 timeout=remaining_time)
```

Comment:
what happens if `remaining_time` is negative here?

Reply:
It crashes! Let me fix that...

Reply:
I feel like `timeout` should only be used for the network stuff. Each command should use the same timeout and not care about the other's performance?

Reply:
I feel like that might be a surprising API, because it would mean that running `upload_and_load_capsule` with a timeout of 5 seconds might end up with a call that could last upwards of 10 seconds.

Reply:
I guess we could make it clear in the docs that the timeout is on a per-network-call basis. That wouldn't be too bad.

Reply:
If I'm loading a webpage, and it relies on loading different content from different places... I don't think it would be intuitive if it timesout if the _entire_ page takes too long, as opposed to a single request hanging everything up?

Reply:
It's a timeout, not a time-allotment

Reply:
Okay, I've switched it over